// @ownego/polaris-vue/dist/es/src/components/FooterHelp/FooterHelp.vue_vue_type_script_setup_true_lang.js
import { defineComponent, useSlots, computed, openBlock, createElementBlock, normalizeClass, unref, normalizeStyle, createVNode, withCtx, renderSlot } from "vue";
import { _ as _sfc_main$1 } from "../Text/Text.vue_vue_type_script_setup_true_lang.js";
const FooterHelp = "Polaris-FooterHelp";
const styles = {
  FooterHelp
};
const _sfc_main = /* @__PURE__ */ defineComponent({
  __name: "FooterHelp",
  props: {
    align: { default: "center" }
  },
  setup(__props) {
    const props = __props;
    useSlots();
    const style = computed(() => ({
      "--pc-footer-help-align": props.align
    }));
    return (_ctx, _cache) => {
      return openBlock(), createElementBlock("div", {
        class: normalizeClass(unref(styles).FooterHelp),
        style: normalizeStyle(style.value)
      }, [
        createVNode(unref(_sfc_main$1), {
          as: "p",
          variant: "bodyLg"
        }, {
          default: withCtx(() => [
            renderSlot(_ctx.$slots, "default")
          ]),
          _: 3
        })
      ], 6);
    };
  }
});
export {
  _sfc_main as _
};
