// @ownego/polaris-vue/dist/es/src/components/IndexFilters/components/SearchFilterButton/SearchFilterButton.vue_vue_type_script_setup_true_lang.js
import { defineComponent, openBlock, createBlock, unref, withCtx, createVNode, createTextVNode, toDisplayString, createElementVNode, normalizeStyle, createCommentVNode } from "vue";
import { S as SearchIcon, F as FilterIcon } from "../../../../../vendor.js";
import { _ as _sfc_main$3 } from "../../../Button/Button.vue_vue_type_script_setup_true_lang.js";
import { _ as _sfc_main$5 } from "../../../Icon/Icon.vue_vue_type_script_setup_true_lang.js";
import { _ as _sfc_main$4 } from "../../../InlineStack/InlineStack.vue_vue_type_script_setup_true_lang.js";
import { _ as _sfc_main$2 } from "../../../Text/Text.vue_vue_type_script_setup_true_lang.js";
import { _ as _sfc_main$1 } from "../../../Tooltip/Tooltip.vue_vue_type_script_setup_true_lang.js";
const _sfc_main = /* @__PURE__ */ defineComponent({
  __name: "SearchFilterButton",
  props: {
    label: {},
    disabled: { type: Boolean },
    tooltipContent: {},
    hideFilters: { type: Boolean },
    hideQueryField: { type: Boolean },
    style: {}
  },
  emits: ["click"],
  setup(__props, { emit: __emit }) {
    const emits = __emit;
    return (_ctx, _cache) => {
      return openBlock(), createBlock(unref(_sfc_main$1), {
        "preferred-position": "above",
        "hover-delay": 400
      }, {
        content: withCtx(() => [
          createVNode(unref(_sfc_main$2), {
            as: "span",
            variant: "bodyMd",
            alignment: "center"
          }, {
            default: withCtx(() => [
              createTextVNode(toDisplayString(_ctx.tooltipContent), 1)
            ]),
            _: 1
          })
        ]),
        default: withCtx(() => [
          createElementVNode("div", {
            style: normalizeStyle(_ctx.style)
          }, [
            createVNode(unref(_sfc_main$3), {
              size: "slim",
              disabled: _ctx.disabled,
              "accessibility-label": _ctx.label,
              onClick: _cache[0] || (_cache[0] = ($event) => emits("click"))
            }, {
              icon: withCtx(() => [
                createVNode(unref(_sfc_main$4), { gap: "0" }, {
                  default: withCtx(() => [
                    !_ctx.hideQueryField ? (openBlock(), createBlock(unref(_sfc_main$5), {
                      key: 0,
                      source: unref(SearchIcon),
                      tone: "base"
                    }, null, 8, ["source"])) : createCommentVNode("", true),
                    !_ctx.hideFilters ? (openBlock(), createBlock(unref(_sfc_main$5), {
                      key: 1,
                      source: unref(FilterIcon),
                      tone: "base"
                    }, null, 8, ["source"])) : createCommentVNode("", true)
                  ]),
                  _: 1
                })
              ]),
              _: 1
            }, 8, ["disabled", "accessibility-label"])
          ], 4)
        ]),
        _: 1
      });
    };
  }
});
export {
  _sfc_main as _
};
