// @ownego/polaris-vue/dist/es/src/components/ActionList/components/Section/Section.vue_vue_type_script_setup_true_lang.js
import { defineComponent, computed, openBlock, createBlock, unref, withCtx, createVNode, mergeProps } from "vue";
import { _ as _sfc_main$2 } from "./SectionMarkUp.vue_vue_type_script_setup_true_lang.js";
import { _ as _sfc_main$1 } from "../../../Box/Box.vue_vue_type_script_setup_true_lang.js";
const _sfc_main = /* @__PURE__ */ defineComponent({
  __name: "Section",
  props: {
    section: {},
    hasMultipleSections: { type: Boolean },
    actionRole: {},
    isFirst: { type: Boolean }
  },
  emits: ["action-any-item"],
  setup(__props, { emit: __emit }) {
    const props = __props;
    const emits = __emit;
    const SectionMarkUpProps = computed(() => {
      return {
        section: props.section,
        hasMultipleSections: props.hasMultipleSections,
        actionRole: props.actionRole,
        isFirst: props.isFirst
      };
    });
    return (_ctx, _cache) => {
      return _ctx.hasMultipleSections ? (openBlock(), createBlock(unref(_sfc_main$1), {
        key: 0,
        as: "li",
        role: "presentation",
        borderColor: "border-secondary",
        "border-block-start-width": !_ctx.isFirst ? "025" : void 0,
        "padding-block-stack": !_ctx.section.title ? "150" : void 0
      }, {
        default: withCtx(() => [
          createVNode(_sfc_main$2, mergeProps(SectionMarkUpProps.value, {
            onActionAnyItem: _cache[0] || (_cache[0] = ($event) => emits("action-any-item"))
          }), null, 16)
        ]),
        _: 1
      }, 8, ["border-block-start-width", "padding-block-stack"])) : (openBlock(), createBlock(_sfc_main$2, mergeProps({ key: 1 }, SectionMarkUpProps.value, {
        onActionAnyItem: _cache[1] || (_cache[1] = ($event) => emits("action-any-item"))
      }), null, 16));
    };
  }
});
export {
  _sfc_main as _
};
