// @ownego/polaris-vue/dist/es/src/components/ActionList/components/Section/SectionMarkUp.vue_vue_type_script_setup_true_lang.js
import { defineComponent, computed, openBlock, createElementBlock, Fragment, createBlock, unref, withCtx, createVNode, createTextVNode, toDisplayString, createCommentVNode, renderList, mergeProps } from "vue";
import { _ as _sfc_main$5 } from "../Item/Item.vue_vue_type_script_setup_true_lang.js";
import { _ as _sfc_main$3 } from "../../../BlockStack/BlockStack.vue_vue_type_script_setup_true_lang.js";
import { _ as _sfc_main$1 } from "../../../Box/Box.vue_vue_type_script_setup_true_lang.js";
import { _ as _sfc_main$4 } from "../../../InlineStack/InlineStack.vue_vue_type_script_setup_true_lang.js";
import { _ as _sfc_main$2 } from "../../../Text/Text.vue_vue_type_script_setup_true_lang.js";
const _sfc_main = /* @__PURE__ */ defineComponent({
  __name: "SectionMarkUp",
  props: {
    section: {},
    hasMultipleSections: { type: Boolean },
    actionRole: {},
    onActionAnyItem: {},
    isFirst: { type: Boolean }
  },
  emits: ["action-any-item"],
  setup(__props, { emit: __emit }) {
    const props = __props;
    const emits = __emit;
    const handleAction = (itemOnAction) => {
      if (itemOnAction) {
        itemOnAction();
      }
      emits("action-any-item");
    };
    const sectionRole = computed(() => {
      if (props.actionRole) {
        if (props.actionRole === "option") {
          return "presentation";
        }
        if (props.actionRole === "menuitem") {
          return !props.hasMultipleSections ? "menu" : "presentation";
        }
        return void 0;
      }
      return void 0;
    });
    return (_ctx, _cache) => {
      return openBlock(), createElementBlock(Fragment, null, [
        _ctx.section.title ? (openBlock(), createElementBlock(Fragment, { key: 0 }, [
          _ctx.section.title ? (openBlock(), createBlock(unref(_sfc_main$1), {
            key: 0,
            paddingBlockStart: "300",
            paddingBlockEnd: "100",
            paddingInlineStart: "300",
            paddingInlineEnd: "300"
          }, {
            default: withCtx(() => [
              createVNode(unref(_sfc_main$2), {
                as: "p",
                variant: "headingSm"
              }, {
                default: withCtx(() => [
                  createTextVNode(toDisplayString(_ctx.section.title), 1)
                ]),
                _: 1
              })
            ]),
            _: 1
          })) : createCommentVNode("", true)
        ], 64)) : createCommentVNode("", true),
        createVNode(unref(_sfc_main$1), {
          as: "div",
          padding: "150",
          "padding-block-stack": _ctx.hasMultipleSections ? "0" : void 0,
          tabIndex: !_ctx.hasMultipleSections ? -1 : void 0
        }, {
          default: withCtx(() => [
            createVNode(unref(_sfc_main$3), {
              gap: "050",
              as: "ul",
              role: sectionRole.value || void 0
            }, {
              default: withCtx(() => [
                (openBlock(true), createElementBlock(Fragment, null, renderList(_ctx.section.items, ({ content, helpText, onAction, ...item }, index) => {
                  return openBlock(), createBlock(unref(_sfc_main$1), {
                    as: "li",
                    key: `${content}-${index}`,
                    role: _ctx.actionRole === "menuitem" ? "presentation" : void 0
                  }, {
                    default: withCtx(() => [
                      createVNode(unref(_sfc_main$4), { wrap: false }, {
                        default: withCtx(() => [
                          createVNode(_sfc_main$5, mergeProps({ ref_for: true }, item, {
                            content,
                            "help-text": helpText,
                            role: _ctx.actionRole,
                            onAction: ($event) => handleAction(onAction)
                          }), null, 16, ["content", "help-text", "role", "onAction"])
                        ]),
                        _: 2
                      }, 1024)
                    ]),
                    _: 2
                  }, 1032, ["role"]);
                }), 128))
              ]),
              _: 1
            }, 8, ["role"])
          ]),
          _: 1
        }, 8, ["padding-block-stack", "tabIndex"])
      ], 64);
    };
  }
});
export {
  _sfc_main as _
};
