// @ownego/polaris-vue/dist/es/src/components/Modal/components/Footer/Footer.vue_vue_type_script_setup_true_lang.js
import { defineComponent, openBlock, createBlock, unref, withCtx, createVNode, renderSlot, createElementBlock, Fragment, renderList, createCommentVNode } from "vue";
import { _ as _sfc_main$2 } from "../../../Box/Box.vue_vue_type_script_setup_true_lang.js";
import { _ as _sfc_main$3 } from "../../../Button/ButtonFrom.vue_vue_type_script_setup_true_lang.js";
import { _ as _sfc_main$1 } from "../../../InlineStack/InlineStack.vue_vue_type_script_setup_true_lang.js";
const _sfc_main = /* @__PURE__ */ defineComponent({
  __name: "Footer",
  props: {
    primaryAction: {},
    secondaryActions: {}
  },
  setup(__props) {
    return (_ctx, _cache) => {
      return openBlock(), createBlock(unref(_sfc_main$1), {
        gap: "400",
        "block-align": "center"
      }, {
        default: withCtx(() => [
          createVNode(unref(_sfc_main$2), {
            "border-color": "border",
            "border-block-start-width": "025",
            padding: "400",
            width: "100%"
          }, {
            default: withCtx(() => [
              createVNode(unref(_sfc_main$1), {
                gap: "400",
                "block-align": "center",
                align: "space-between"
              }, {
                default: withCtx(() => [
                  createVNode(unref(_sfc_main$2), null, {
                    default: withCtx(() => [
                      renderSlot(_ctx.$slots, "default")
                    ]),
                    _: 3
                  }),
                  _ctx.primaryAction || _ctx.secondaryActions ? (openBlock(), createBlock(unref(_sfc_main$1), {
                    key: 0,
                    gap: "200"
                  }, {
                    default: withCtx(() => [
                      _ctx.secondaryActions ? (openBlock(true), createElementBlock(Fragment, { key: 0 }, renderList(_ctx.secondaryActions, (action, index) => {
                        return openBlock(), createBlock(unref(_sfc_main$3), {
                          key: String(index),
                          action
                        }, null, 8, ["action"]);
                      }), 128)) : createCommentVNode("", true),
                      _ctx.primaryAction ? (openBlock(), createBlock(unref(_sfc_main$3), {
                        key: 1,
                        action: _ctx.primaryAction,
                        overrides: { variant: "primary" }
                      }, null, 8, ["action"])) : createCommentVNode("", true)
                    ]),
                    _: 1
                  })) : createCommentVNode("", true)
                ]),
                _: 3
              })
            ]),
            _: 3
          })
        ]),
        _: 3
      });
    };
  }
});
export {
  _sfc_main as _
};
