// @ownego/polaris-vue/dist/es/src/components/Grid/components/Cell/Cell.vue_vue_type_script_setup_true_lang.js
import { defineComponent, computed, openBlock, createElementBlock, normalizeClass, normalizeStyle, renderSlot } from "vue";
import { c as classNames } from "../../../../utilities/css.js";
const Cell = "Polaris-Grid-Cell";
const styles = {
  Cell,
  "Cell-1-column-xs": "Polaris-Grid-Cell--cell_1ColumnXs",
  "Cell-2-column-xs": "Polaris-Grid-Cell--cell_2ColumnXs",
  "Cell-3-column-xs": "Polaris-Grid-Cell--cell_3ColumnXs",
  "Cell-4-column-xs": "Polaris-Grid-Cell--cell_4ColumnXs",
  "Cell-5-column-xs": "Polaris-Grid-Cell--cell_5ColumnXs",
  "Cell-6-column-xs": "Polaris-Grid-Cell--cell_6ColumnXs",
  "Cell-1-column-sm": "Polaris-Grid-Cell--cell_1ColumnSm",
  "Cell-2-column-sm": "Polaris-Grid-Cell--cell_2ColumnSm",
  "Cell-3-column-sm": "Polaris-Grid-Cell--cell_3ColumnSm",
  "Cell-4-column-sm": "Polaris-Grid-Cell--cell_4ColumnSm",
  "Cell-5-column-sm": "Polaris-Grid-Cell--cell_5ColumnSm",
  "Cell-6-column-sm": "Polaris-Grid-Cell--cell_6ColumnSm",
  "Cell-1-column-md": "Polaris-Grid-Cell--cell_1ColumnMd",
  "Cell-2-column-md": "Polaris-Grid-Cell--cell_2ColumnMd",
  "Cell-3-column-md": "Polaris-Grid-Cell--cell_3ColumnMd",
  "Cell-4-column-md": "Polaris-Grid-Cell--cell_4ColumnMd",
  "Cell-5-column-md": "Polaris-Grid-Cell--cell_5ColumnMd",
  "Cell-6-column-md": "Polaris-Grid-Cell--cell_6ColumnMd",
  "Cell-1-column-lg": "Polaris-Grid-Cell--cell_1ColumnLg",
  "Cell-2-column-lg": "Polaris-Grid-Cell--cell_2ColumnLg",
  "Cell-3-column-lg": "Polaris-Grid-Cell--cell_3ColumnLg",
  "Cell-4-column-lg": "Polaris-Grid-Cell--cell_4ColumnLg",
  "Cell-5-column-lg": "Polaris-Grid-Cell--cell_5ColumnLg",
  "Cell-6-column-lg": "Polaris-Grid-Cell--cell_6ColumnLg",
  "Cell-7-column-lg": "Polaris-Grid-Cell--cell_7ColumnLg",
  "Cell-8-column-lg": "Polaris-Grid-Cell--cell_8ColumnLg",
  "Cell-9-column-lg": "Polaris-Grid-Cell--cell_9ColumnLg",
  "Cell-10-column-lg": "Polaris-Grid-Cell--cell_10ColumnLg",
  "Cell-11-column-lg": "Polaris-Grid-Cell--cell_11ColumnLg",
  "Cell-12-column-lg": "Polaris-Grid-Cell--cell_12ColumnLg",
  "Cell-1-column-xl": "Polaris-Grid-Cell--cell_1ColumnXl",
  "Cell-2-column-xl": "Polaris-Grid-Cell--cell_2ColumnXl",
  "Cell-3-column-xl": "Polaris-Grid-Cell--cell_3ColumnXl",
  "Cell-4-column-xl": "Polaris-Grid-Cell--cell_4ColumnXl",
  "Cell-5-column-xl": "Polaris-Grid-Cell--cell_5ColumnXl",
  "Cell-6-column-xl": "Polaris-Grid-Cell--cell_6ColumnXl",
  "Cell-7-column-xl": "Polaris-Grid-Cell--cell_7ColumnXl",
  "Cell-8-column-xl": "Polaris-Grid-Cell--cell_8ColumnXl",
  "Cell-9-column-xl": "Polaris-Grid-Cell--cell_9ColumnXl",
  "Cell-10-column-xl": "Polaris-Grid-Cell--cell_10ColumnXl",
  "Cell-11-column-xl": "Polaris-Grid-Cell--cell_11ColumnXl",
  "Cell-12-column-xl": "Polaris-Grid-Cell--cell_12ColumnXl"
};
const _sfc_main = /* @__PURE__ */ defineComponent({
  __name: "Cell",
  props: {
    area: {},
    column: {},
    columnSpan: {},
    row: {}
  },
  setup(__props) {
    const props = __props;
    const className = computed(() => {
      var _a, _b, _c, _d, _e;
      return classNames(
        styles.Cell,
        ((_a = props.columnSpan) == null ? void 0 : _a.xs) && styles[`Cell-${props.columnSpan.xs}-column-xs`],
        ((_b = props.columnSpan) == null ? void 0 : _b.sm) && styles[`Cell-${props.columnSpan.sm}-column-sm`],
        ((_c = props.columnSpan) == null ? void 0 : _c.md) && styles[`Cell-${props.columnSpan.md}-column-md`],
        ((_d = props.columnSpan) == null ? void 0 : _d.lg) && styles[`Cell-${props.columnSpan.lg}-column-lg`],
        ((_e = props.columnSpan) == null ? void 0 : _e.xl) && styles[`Cell-${props.columnSpan.xl}-column-xl`]
      );
    });
    const style = computed(() => {
      var _a, _b, _c, _d, _e, _f, _g, _h, _i, _j;
      return {
        "gridArea": props.area,
        "--pc-column-xs": (_a = props.column) == null ? void 0 : _a.xs,
        "--pc-column-sm": (_b = props.column) == null ? void 0 : _b.sm,
        "--pc-column-md": (_c = props.column) == null ? void 0 : _c.md,
        "--pc-column-lg": (_d = props.column) == null ? void 0 : _d.lg,
        "--pc-column-xl": (_e = props.column) == null ? void 0 : _e.xl,
        "--pc-row-xs": (_f = props.row) == null ? void 0 : _f.xs,
        "--pc-row-sm": (_g = props.row) == null ? void 0 : _g.sm,
        "--pc-row-md": (_h = props.row) == null ? void 0 : _h.md,
        "--pc-row-lg": (_i = props.row) == null ? void 0 : _i.lg,
        "--pc-row-xl": (_j = props.row) == null ? void 0 : _j.xl
      };
    });
    return (_ctx, _cache) => {
      return openBlock(), createElementBlock("div", {
        class: normalizeClass(className.value),
        style: normalizeStyle(style.value)
      }, [
        renderSlot(_ctx.$slots, "default")
      ], 6);
    };
  }
});
export {
  _sfc_main as _
};
