// @ownego/polaris-vue/dist/es/src/components/Layout/components/AnnotatedSection/AnnotatedSection.vue_vue_type_script_setup_true_lang.js
import { defineComponent, useSlots, computed, openBlock, createElementBlock, normalizeClass, unref, createElementVNode, createVNode, withCtx, renderSlot, Fragment, createTextVNode, toDisplayString, createBlock, createCommentVNode } from "vue";
import { _ as _sfc_main$3 } from "../../../Box/Box.vue_vue_type_script_setup_true_lang.js";
import { s as styles } from "../../Layout.vue_vue_type_script_setup_true_lang.js";
import { _ as _sfc_main$2 } from "../../../Text/Text.vue_vue_type_script_setup_true_lang.js";
import { _ as _sfc_main$1 } from "../../../TextContainer/TextContainer.vue_vue_type_script_setup_true_lang.js";
const _sfc_main = /* @__PURE__ */ defineComponent({
  __name: "AnnotatedSection",
  props: {
    id: {},
    title: {},
    description: {}
  },
  setup(__props) {
    const props = __props;
    const slots = useSlots();
    const hasDescription = computed(() => Boolean(slots.description || props.description));
    return (_ctx, _cache) => {
      return openBlock(), createElementBlock("div", {
        class: normalizeClass(unref(styles).AnnotatedSection)
      }, [
        createElementVNode("div", {
          class: normalizeClass(unref(styles).AnnotationWrapper)
        }, [
          createElementVNode("div", {
            class: normalizeClass(unref(styles).Annotation)
          }, [
            createVNode(unref(_sfc_main$1), { spacing: "tight" }, {
              default: withCtx(() => [
                createVNode(unref(_sfc_main$2), {
                  id: _ctx.id,
                  variant: "headingMd",
                  as: "h2"
                }, {
                  default: withCtx(() => [
                    slots.title ? renderSlot(_ctx.$slots, "title", { key: 0 }) : (openBlock(), createElementBlock(Fragment, { key: 1 }, [
                      createTextVNode(toDisplayString(_ctx.title), 1)
                    ], 64))
                  ]),
                  _: 3
                }, 8, ["id"]),
                hasDescription.value ? (openBlock(), createBlock(unref(_sfc_main$3), {
                  key: 0,
                  color: "text-secondary"
                }, {
                  default: withCtx(() => [
                    slots.description ? renderSlot(_ctx.$slots, "description", { key: 0 }) : (openBlock(), createBlock(unref(_sfc_main$2), {
                      key: 1,
                      as: "p",
                      variant: "bodyMd"
                    }, {
                      default: withCtx(() => [
                        createTextVNode(toDisplayString(_ctx.description), 1)
                      ]),
                      _: 1
                    }))
                  ]),
                  _: 3
                })) : createCommentVNode("", true)
              ]),
              _: 3
            })
          ], 2),
          createElementVNode("div", {
            class: normalizeClass(unref(styles).AnnotationContent)
          }, [
            renderSlot(_ctx.$slots, "default")
          ], 2)
        ], 2)
      ], 2);
    };
  }
});
export {
  _sfc_main as _
};
