// @ownego/polaris-vue/dist/es/src/components/ExceptionList/ExceptionList.vue_vue_type_script_setup_true_lang.js
import { defineComponent, openBlock, createElementBlock, normalizeClass, unref, Fragment, renderList, createElementVNode, createBlock, withCtx, toDisplayString, createCommentVNode } from "vue";
import { c as classNames, v as variationName } from "../../utilities/css.js";
import { _ as _sfc_main$1 } from "../Icon/Icon.vue_vue_type_script_setup_true_lang.js";
import { _ as _sfc_main$2 } from "../Truncate/Truncate.vue_vue_type_script_setup_true_lang.js";
const ExceptionList = "Polaris-ExceptionList";
const Item = "Polaris-ExceptionList__Item";
const Icon = "Polaris-ExceptionList__Icon";
const statusWarning = "Polaris-ExceptionList--statusWarning";
const statusCritical = "Polaris-ExceptionList--statusCritical";
const Bullet = "Polaris-ExceptionList__Bullet";
const Title = "Polaris-ExceptionList__Title";
const Description = "Polaris-ExceptionList__Description";
const styles = {
  ExceptionList,
  Item,
  Icon,
  statusWarning,
  statusCritical,
  Bullet,
  Title,
  Description
};
const _sfc_main = /* @__PURE__ */ defineComponent({
  __name: "ExceptionList",
  props: {
    items: {}
  },
  setup(__props) {
    const itemClasses = (item) => classNames(
      styles.Item,
      item.status && styles[variationName("status", item.status)]
    );
    return (_ctx, _cache) => {
      return openBlock(), createElementBlock("ul", {
        class: normalizeClass(unref(styles).ExceptionList)
      }, [
        (openBlock(true), createElementBlock(Fragment, null, renderList(_ctx.items, (item, index) => {
          return openBlock(), createElementBlock("li", {
            key: index,
            class: normalizeClass(itemClasses(item))
          }, [
            createElementVNode("span", {
              class: normalizeClass(unref(styles).Icon)
            }, [
              item.icon ? (openBlock(), createBlock(unref(_sfc_main$1), {
                key: 0,
                source: item.icon
              }, null, 8, ["source"])) : (openBlock(), createElementBlock("span", {
                key: 1,
                class: normalizeClass(unref(styles).Bullet)
              }, null, 2))
            ], 2),
            item.truncate ? (openBlock(), createBlock(unref(_sfc_main$2), { key: 0 }, {
              default: withCtx(() => [
                item.title ? (openBlock(), createElementBlock("span", {
                  key: 0,
                  class: normalizeClass(unref(styles).Title)
                }, toDisplayString(item.title), 3)) : createCommentVNode("", true),
                item.description ? (openBlock(), createElementBlock("span", {
                  key: 1,
                  class: normalizeClass(unref(styles).Description)
                }, toDisplayString(item.description), 3)) : createCommentVNode("", true)
              ]),
              _: 2
            }, 1024)) : (openBlock(), createElementBlock(Fragment, { key: 1 }, [
              item.title ? (openBlock(), createElementBlock("span", {
                key: 0,
                class: normalizeClass(unref(styles).Title)
              }, toDisplayString(item.title), 3)) : createCommentVNode("", true),
              item.description ? (openBlock(), createElementBlock("span", {
                key: 1,
                class: normalizeClass(unref(styles).Description)
              }, toDisplayString(item.description), 3)) : createCommentVNode("", true)
            ], 64))
          ], 2);
        }), 128))
      ], 2);
    };
  }
});
export {
  _sfc_main as _
};
