// @ownego/polaris-vue/dist/es/src/components/SkeletonTabs/SkeletonTabs.vue_vue_type_script_setup_true_lang.js
import { defineComponent, openBlock, createElementBlock, normalizeClass, unref, Fragment, renderList, createElementVNode } from "vue";
import { c as classNames } from "../../utilities/css.js";
const Tabs = "Polaris-SkeletonTabs__Tabs";
const Tab = "Polaris-SkeletonTabs__Tab";
const TabText = "Polaris-SkeletonTabs__TabText";
const fitted = "Polaris-SkeletonTabs--fitted";
const styles = {
  Tabs,
  Tab,
  TabText,
  fitted
};
const _sfc_main = /* @__PURE__ */ defineComponent({
  __name: "SkeletonTabs",
  props: {
    count: { default: 2 },
    fitted: { type: Boolean, default: false }
  },
  setup(__props) {
    return (_ctx, _cache) => {
      return openBlock(), createElementBlock("div", {
        class: normalizeClass(unref(classNames)(unref(styles).Tabs, _ctx.fitted && unref(styles).fitted))
      }, [
        (openBlock(true), createElementBlock(Fragment, null, renderList(_ctx.count, (index) => {
          return openBlock(), createElementBlock("div", {
            key: index,
            class: normalizeClass(unref(classNames)(unref(styles).Tab))
          }, [
            createElementVNode("div", {
              class: normalizeClass(unref(styles).TabText)
            }, null, 2)
          ], 2);
        }), 128))
      ], 2);
    };
  }
});
export {
  _sfc_main as _
};
