// @ownego/polaris-vue/dist/es/src/components/EmptySearchResult/EmptySearchResult.vue_vue_type_script_setup_true_lang.js
import { defineComponent, openBlock, createBlock, unref, withCtx, createCommentVNode, createVNode, createTextVNode, toDisplayString, createElementBlock } from "vue";
import { u as useI18n } from "../../use/useI18n.js";
import { _ as _sfc_main$2 } from "../Image/Image.vue_vue_type_script_setup_true_lang.js";
import { _ as _sfc_main$1 } from "../LegacyStack/LegacyStack.vue_vue_type_script_setup_true_lang.js";
import { _ as _sfc_main$3 } from "../Text/Text.vue_vue_type_script_setup_true_lang.js";
const _hoisted_1 = { key: 0 };
const source = "data:image/svg+xml,%3csvg width='60' height='60' xmlns='http://www.w3.org/2000/svg'%3e%3cpath fill-rule='evenodd' d='M41.87 24a17.87 17.87 0 11-35.74 0 17.87 17.87 0 0135.74 0zm-3.15 18.96a24 24 0 114.24-4.24L59.04 54.8a3 3 0 11-4.24 4.24L38.72 42.96z' fill='%238C9196'/%3e%3c/svg%3e";
const _sfc_main = /* @__PURE__ */ defineComponent({
  __name: "EmptySearchResult",
  props: {
    title: {},
    description: {},
    withIllustration: { type: Boolean }
  },
  setup(__props) {
    const i18n = useI18n();
    return (_ctx, _cache) => {
      return openBlock(), createBlock(unref(_sfc_main$1), {
        alignment: "center",
        vertical: ""
      }, {
        default: withCtx(() => [
          _ctx.withIllustration ? (openBlock(), createBlock(unref(_sfc_main$2), {
            key: 0,
            alt: unref(i18n).translate("Polaris.EmptySearchResult.altText"),
            source,
            draggable: false
          }, null, 8, ["alt"])) : createCommentVNode("", true),
          createVNode(unref(_sfc_main$3), {
            variant: "headingLg",
            as: "p"
          }, {
            default: withCtx(() => [
              createTextVNode(toDisplayString(_ctx.title), 1)
            ]),
            _: 1
          }),
          createVNode(unref(_sfc_main$3), {
            tone: "subdued",
            as: "span"
          }, {
            default: withCtx(() => [
              _ctx.description ? (openBlock(), createElementBlock("p", _hoisted_1, toDisplayString(_ctx.description), 1)) : createCommentVNode("", true)
            ]),
            _: 1
          })
        ]),
        _: 1
      });
    };
  }
});
export {
  _sfc_main as _
};
