// @ownego/polaris-vue/dist/es/src/components/LegacyCard/LegacyCard.vue_vue_type_script_setup_true_lang.js
import { defineComponent, useSlots, computed, provide, openBlock, createElementBlock, normalizeClass, unref, createBlock, withCtx, renderSlot, createCommentVNode, createVNode, createTextVNode, toDisplayString } from "vue";
import { u as useI18n } from "../../use/useI18n.js";
import { u as useToggle } from "../../use/useToggle.js";
import { u as useLegacyCardPaddingObserverRef } from "../../use/useLegacyCard.js";
import { c as classNames } from "../../utilities/css.js";
import { u as useHasSlot } from "../../use/useHasSlot.js";
import { _ as _sfc_main$7 } from "../ActionList/ActionList.vue_vue_type_script_setup_true_lang.js";
import { _ as _sfc_main$6 } from "../Button/Button.vue_vue_type_script_setup_true_lang.js";
import { _ as _sfc_main$4 } from "../Button/ButtonFrom.vue_vue_type_script_setup_true_lang.js";
import { _ as _sfc_main$3 } from "../ButtonGroup/ButtonGroup.vue_vue_type_script_setup_true_lang.js";
import { _ as _sfc_main$1 } from "./components/Header/Header.vue_vue_type_script_setup_true_lang.js";
import { _ as _sfc_main$2 } from "./components/Section/Section.vue_vue_type_script_setup_true_lang.js";
import { _ as _sfc_main$5 } from "../Popover/Popover.vue_vue_type_script_setup_true_lang.js";
const LegacyCard = "Polaris-LegacyCard";
const Section = "Polaris-LegacyCard__Section";
const subdued = "Polaris-LegacyCard--subdued";
const hideOnPrint = "Polaris-LegacyCard--hideOnPrint";
const Header = "Polaris-LegacyCard__Header";
const SectionHeader = "Polaris-LegacyCard__SectionHeader";
const Subsection = "Polaris-LegacyCard__Subsection";
const Footer = "Polaris-LegacyCard__Footer";
const LeftJustified = "Polaris-LegacyCard__LeftJustified";
const FirstSectionPadding = "Polaris-LegacyCard__FirstSectionPadding";
const LastSectionPadding = "Polaris-LegacyCard__LastSectionPadding";
const styles = {
  LegacyCard,
  Section,
  subdued,
  "Section-hideOnPrint": "Polaris-LegacyCard__Section--hideOnPrint",
  hideOnPrint,
  Header,
  "Section-fullWidth": "Polaris-LegacyCard__Section--fullWidth",
  "Section-flush": "Polaris-LegacyCard__Section--flush",
  "Section-subdued": "Polaris-LegacyCard__Section--subdued",
  SectionHeader,
  Subsection,
  Footer,
  LeftJustified,
  FirstSectionPadding,
  LastSectionPadding
};
const _sfc_main = /* @__PURE__ */ defineComponent({
  __name: "LegacyCard",
  props: {
    title: {},
    subdued: { type: Boolean },
    sectioned: { type: Boolean },
    actions: {},
    primaryFooterAction: {},
    secondaryFooterActions: {},
    secondaryFooterActionsDisclosureText: {},
    footerActionAlignment: { default: "right" },
    hideOnPrint: { type: Boolean }
  },
  setup(__props) {
    const props = __props;
    const i18n = useI18n();
    const slots = useSlots();
    const legacyCardRef = useLegacyCardPaddingObserverRef();
    const { hasSlot } = useHasSlot();
    const {
      value: secondaryActionsPopoverOpen,
      toggle: toggleSecondaryActionsPopoverOpen
    } = useToggle(false);
    const className = computed(
      () => classNames(
        styles.LegacyCard,
        props.subdued && styles.subdued,
        props.hideOnPrint && styles.hideOnPrint
      )
    );
    const footerMarkupStyle = computed(
      () => classNames(
        styles.Footer,
        props.footerActionAlignment === "left" && styles.LeftJustified
      )
    );
    provide("within-content-context", true);
    return (_ctx, _cache) => {
      return openBlock(), createElementBlock("div", {
        ref_key: "legacyCardRef",
        ref: legacyCardRef,
        class: normalizeClass(className.value)
      }, [
        _ctx.title || unref(hasSlot)(unref(slots).title) || _ctx.actions ? (openBlock(), createBlock(unref(_sfc_main$1), {
          key: 0,
          actions: _ctx.actions,
          title: _ctx.title
        }, {
          default: withCtx(() => [
            unref(hasSlot)(unref(slots).title) ? renderSlot(_ctx.$slots, "title", { key: 0 }) : createCommentVNode("", true)
          ]),
          _: 3
        }, 8, ["actions", "title"])) : createCommentVNode("", true),
        _ctx.sectioned ? (openBlock(), createBlock(unref(_sfc_main$2), { key: 1 }, {
          default: withCtx(() => [
            renderSlot(_ctx.$slots, "default")
          ]),
          _: 3
        })) : renderSlot(_ctx.$slots, "default", { key: 2 }),
        _ctx.primaryFooterAction || _ctx.secondaryFooterActions && _ctx.secondaryFooterActions.length ? (openBlock(), createElementBlock("div", {
          key: 3,
          class: normalizeClass(footerMarkupStyle.value)
        }, [
          _ctx.footerActionAlignment === "right" ? (openBlock(), createBlock(unref(_sfc_main$3), { key: 0 }, {
            default: withCtx(() => [
              _ctx.secondaryFooterActions && _ctx.secondaryFooterActions.length === 1 ? (openBlock(), createBlock(unref(_sfc_main$4), {
                key: 0,
                action: _ctx.secondaryFooterActions[0]
              }, null, 8, ["action"])) : (openBlock(), createBlock(unref(_sfc_main$5), {
                key: 1,
                active: unref(secondaryActionsPopoverOpen),
                onClose: unref(toggleSecondaryActionsPopoverOpen)
              }, {
                activator: withCtx(() => [
                  createVNode(unref(_sfc_main$6), {
                    disclosure: true,
                    onClick: unref(toggleSecondaryActionsPopoverOpen)
                  }, {
                    default: withCtx(() => [
                      createTextVNode(toDisplayString(_ctx.secondaryFooterActionsDisclosureText || unref(i18n).translate("Polaris.Common.more")), 1)
                    ]),
                    _: 1
                  }, 8, ["onClick"])
                ]),
                default: withCtx(() => [
                  createVNode(unref(_sfc_main$7), { items: _ctx.secondaryFooterActions }, null, 8, ["items"])
                ]),
                _: 1
              }, 8, ["active", "onClose"])),
              _ctx.primaryFooterAction ? (openBlock(), createBlock(unref(_sfc_main$4), {
                key: 2,
                action: _ctx.primaryFooterAction,
                overrides: { variant: "primary" }
              }, null, 8, ["action"])) : createCommentVNode("", true)
            ]),
            _: 1
          })) : (openBlock(), createBlock(unref(_sfc_main$3), { key: 1 }, {
            default: withCtx(() => [
              _ctx.primaryFooterAction ? (openBlock(), createBlock(unref(_sfc_main$4), {
                key: 0,
                action: _ctx.primaryFooterAction,
                overrides: { variant: "primary" }
              }, null, 8, ["action"])) : createCommentVNode("", true),
              _ctx.secondaryFooterActions && _ctx.secondaryFooterActions.length === 1 ? (openBlock(), createBlock(unref(_sfc_main$4), {
                key: 1,
                action: _ctx.secondaryFooterActions[0]
              }, null, 8, ["action"])) : (openBlock(), createBlock(unref(_sfc_main$5), {
                key: 2,
                active: unref(secondaryActionsPopoverOpen),
                onClose: unref(toggleSecondaryActionsPopoverOpen)
              }, {
                activator: withCtx(() => [
                  createVNode(unref(_sfc_main$6), {
                    disclosure: true,
                    onClick: unref(toggleSecondaryActionsPopoverOpen)
                  }, {
                    default: withCtx(() => [
                      createTextVNode(toDisplayString(_ctx.secondaryFooterActionsDisclosureText || unref(i18n).translate("Polaris.Common.more")), 1)
                    ]),
                    _: 1
                  }, 8, ["onClick"])
                ]),
                default: withCtx(() => [
                  createVNode(unref(_sfc_main$7), { items: _ctx.secondaryFooterActions }, null, 8, ["items"])
                ]),
                _: 1
              }, 8, ["active", "onClose"]))
            ]),
            _: 1
          }))
        ], 2)) : createCommentVNode("", true)
      ], 2);
    };
  }
});
export {
  _sfc_main as _,
  styles as s
};
