// @ownego/polaris-vue/dist/es/src/components/PageActions/PageActions.vue_vue_type_script_setup_true_lang.js
import { defineComponent, useSlots, openBlock, createElementBlock, normalizeClass, unref, createVNode, withCtx, renderSlot, createBlock, Fragment, renderList, createCommentVNode } from "vue";
import { u as useHasSlot } from "../../use/useHasSlot.js";
import { _ as _sfc_main$3 } from "../Button/ButtonFrom.vue_vue_type_script_setup_true_lang.js";
import { _ as _sfc_main$2 } from "../ButtonGroup/ButtonGroup.vue_vue_type_script_setup_true_lang.js";
import { _ as _sfc_main$1 } from "../LegacyStack/LegacyStack.vue_vue_type_script_setup_true_lang.js";
const PageActions = "Polaris-PageActions";
const styles = {
  PageActions
};
const _sfc_main = /* @__PURE__ */ defineComponent({
  __name: "PageActions",
  props: {
    primaryAction: {},
    secondaryActions: {}
  },
  setup(__props) {
    const slots = useSlots();
    const { hasSlot } = useHasSlot();
    return (_ctx, _cache) => {
      return openBlock(), createElementBlock("div", {
        class: normalizeClass(unref(styles).PageActions)
      }, [
        createVNode(unref(_sfc_main$1), {
          distribution: "trailing",
          spacing: "tight"
        }, {
          default: withCtx(() => [
            unref(hasSlot)(slots.secondaryActions) ? renderSlot(_ctx.$slots, "secondaryActions", { key: 0 }) : _ctx.secondaryActions && _ctx.secondaryActions.length > 0 ? (openBlock(), createBlock(unref(_sfc_main$2), { key: 1 }, {
              default: withCtx(() => [
                (openBlock(true), createElementBlock(Fragment, null, renderList(_ctx.secondaryActions, (action, index) => {
                  return openBlock(), createBlock(unref(_sfc_main$3), {
                    key: action.id || `${index}`,
                    action
                  }, null, 8, ["action"]);
                }), 128))
              ]),
              _: 1
            })) : createCommentVNode("", true),
            unref(hasSlot)(slots.primaryAction) ? renderSlot(_ctx.$slots, "primaryAction", { key: 2 }) : _ctx.primaryAction ? (openBlock(), createBlock(unref(_sfc_main$3), {
              key: 3,
              action: _ctx.primaryAction,
              overrides: { variant: "primary" }
            }, null, 8, ["action"])) : createCommentVNode("", true)
          ]),
          _: 3
        })
      ], 2);
    };
  }
});
export {
  _sfc_main as _
};
