// @ownego/polaris-vue/dist/es/src/components/DataTable/components/Navigation/Navigation.vue_vue_type_script_setup_true_lang.js
import { defineComponent, ref, openBlock, createElementBlock, normalizeClass, unref, createVNode, Fragment, renderList, createCommentVNode } from "vue";
import { c as classNames } from "../../../../utilities/css.js";
import { u as useI18n } from "../../../../use/useI18n.js";
import { _ as _sfc_main$1 } from "../../../Button/Button.vue_vue_type_script_setup_true_lang.js";
import { s as styles } from "../../DataTable.vue_vue_type_script_setup_true_lang.js";
import { m as ChevronLeftIcon, n as ChevronRightIcon } from "../../../../../vendor.js";
const _sfc_main = /* @__PURE__ */ defineComponent({
  __name: "Navigation",
  props: {
    columnVisibilityData: {},
    isScrolledFarthestLeft: { type: Boolean },
    isScrolledFarthestRight: { type: Boolean },
    fixedFirstColumns: {},
    setRef: { type: Function }
  },
  emits: ["navigate-table-left", "navigate-table-right"],
  setup(__props, { expose: __expose, emit: __emit }) {
    const emits = __emit;
    const i18n = useI18n();
    const navigationRef = ref(null);
    const leftA11yLabel = i18n.translate(
      "Polaris.DataTable.navAccessibilityLabel",
      { direction: "left" }
    );
    const rightA11yLabel = i18n.translate(
      "Polaris.DataTable.navAccessibilityLabel",
      { direction: "right" }
    );
    __expose({ el: navigationRef });
    return (_ctx, _cache) => {
      return openBlock(), createElementBlock("div", {
        class: normalizeClass(unref(styles).Navigation),
        ref: _ctx.setRef
      }, [
        createVNode(unref(_sfc_main$1), {
          plain: "",
          icon: unref(ChevronLeftIcon),
          disabled: _ctx.isScrolledFarthestLeft,
          accessibilityLabel: unref(leftA11yLabel),
          onClick: _cache[0] || (_cache[0] = ($event) => emits("navigate-table-left"))
        }, null, 8, ["icon", "disabled", "accessibilityLabel"]),
        (openBlock(true), createElementBlock(Fragment, null, renderList(_ctx.columnVisibilityData, (column, index) => {
          return openBlock(), createElementBlock(Fragment, {
            key: `pip-${index}`
          }, [
            _ctx.fixedFirstColumns <= index ? (openBlock(), createElementBlock("div", {
              key: 0,
              class: normalizeClass(unref(classNames)(unref(styles).Pip, column.isVisible && unref(styles)["Pip-visible"]))
            }, null, 2)) : createCommentVNode("", true)
          ], 64);
        }), 128)),
        createVNode(unref(_sfc_main$1), {
          plain: "",
          icon: unref(ChevronRightIcon),
          disabled: _ctx.isScrolledFarthestRight,
          accessibilityLabel: unref(rightA11yLabel),
          onClick: _cache[1] || (_cache[1] = ($event) => emits("navigate-table-right"))
        }, null, 8, ["icon", "disabled", "accessibilityLabel"])
      ], 2);
    };
  }
});
export {
  _sfc_main as _
};
