// @ownego/polaris-vue/dist/es/src/utilities/types.js
var Key = /* @__PURE__ */ ((Key10) => {
  Key10[Key10["Backspace"] = 8] = "Backspace";
  Key10[Key10["Tab"] = 9] = "Tab";
  Key10[Key10["Enter"] = 13] = "Enter";
  Key10[Key10["Shift"] = 16] = "Shift";
  Key10[Key10["Ctrl"] = 17] = "Ctrl";
  Key10[Key10["Alt"] = 18] = "Alt";
  Key10[Key10["Pause"] = 19] = "Pause";
  Key10[Key10["CapsLock"] = 20] = "CapsLock";
  Key10[Key10["Escape"] = 27] = "Escape";
  Key10[Key10["Space"] = 32] = "Space";
  Key10[Key10["PageUp"] = 33] = "PageUp";
  Key10[Key10["PageDown"] = 34] = "PageDown";
  Key10[Key10["End"] = 35] = "End";
  Key10[Key10["Home"] = 36] = "Home";
  Key10[Key10["LeftArrow"] = 37] = "LeftArrow";
  Key10[Key10["UpArrow"] = 38] = "UpArrow";
  Key10[Key10["RightArrow"] = 39] = "RightArrow";
  Key10[Key10["DownArrow"] = 40] = "DownArrow";
  Key10[Key10["Insert"] = 45] = "Insert";
  Key10[Key10["Delete"] = 46] = "Delete";
  Key10[Key10["Key0"] = 48] = "Key0";
  Key10[Key10["Key1"] = 49] = "Key1";
  Key10[Key10["Key2"] = 50] = "Key2";
  Key10[Key10["Key3"] = 51] = "Key3";
  Key10[Key10["Key4"] = 52] = "Key4";
  Key10[Key10["Key5"] = 53] = "Key5";
  Key10[Key10["Key6"] = 54] = "Key6";
  Key10[Key10["Key7"] = 55] = "Key7";
  Key10[Key10["Key8"] = 56] = "Key8";
  Key10[Key10["Key9"] = 57] = "Key9";
  Key10[Key10["KeyA"] = 65] = "KeyA";
  Key10[Key10["KeyB"] = 66] = "KeyB";
  Key10[Key10["KeyC"] = 67] = "KeyC";
  Key10[Key10["KeyD"] = 68] = "KeyD";
  Key10[Key10["KeyE"] = 69] = "KeyE";
  Key10[Key10["KeyF"] = 70] = "KeyF";
  Key10[Key10["KeyG"] = 71] = "KeyG";
  Key10[Key10["KeyH"] = 72] = "KeyH";
  Key10[Key10["KeyI"] = 73] = "KeyI";
  Key10[Key10["KeyJ"] = 74] = "KeyJ";
  Key10[Key10["KeyK"] = 75] = "KeyK";
  Key10[Key10["KeyL"] = 76] = "KeyL";
  Key10[Key10["KeyM"] = 77] = "KeyM";
  Key10[Key10["KeyN"] = 78] = "KeyN";
  Key10[Key10["KeyO"] = 79] = "KeyO";
  Key10[Key10["KeyP"] = 80] = "KeyP";
  Key10[Key10["KeyQ"] = 81] = "KeyQ";
  Key10[Key10["KeyR"] = 82] = "KeyR";
  Key10[Key10["KeyS"] = 83] = "KeyS";
  Key10[Key10["KeyT"] = 84] = "KeyT";
  Key10[Key10["KeyU"] = 85] = "KeyU";
  Key10[Key10["KeyV"] = 86] = "KeyV";
  Key10[Key10["KeyW"] = 87] = "KeyW";
  Key10[Key10["KeyX"] = 88] = "KeyX";
  Key10[Key10["KeyY"] = 89] = "KeyY";
  Key10[Key10["KeyZ"] = 90] = "KeyZ";
  Key10[Key10["LeftMeta"] = 91] = "LeftMeta";
  Key10[Key10["RightMeta"] = 92] = "RightMeta";
  Key10[Key10["Select"] = 93] = "Select";
  Key10[Key10["Numpad0"] = 96] = "Numpad0";
  Key10[Key10["Numpad1"] = 97] = "Numpad1";
  Key10[Key10["Numpad2"] = 98] = "Numpad2";
  Key10[Key10["Numpad3"] = 99] = "Numpad3";
  Key10[Key10["Numpad4"] = 100] = "Numpad4";
  Key10[Key10["Numpad5"] = 101] = "Numpad5";
  Key10[Key10["Numpad6"] = 102] = "Numpad6";
  Key10[Key10["Numpad7"] = 103] = "Numpad7";
  Key10[Key10["Numpad8"] = 104] = "Numpad8";
  Key10[Key10["Numpad9"] = 105] = "Numpad9";
  Key10[Key10["Multiply"] = 106] = "Multiply";
  Key10[Key10["Add"] = 107] = "Add";
  Key10[Key10["Subtract"] = 109] = "Subtract";
  Key10[Key10["Decimal"] = 110] = "Decimal";
  Key10[Key10["Divide"] = 111] = "Divide";
  Key10[Key10["F1"] = 112] = "F1";
  Key10[Key10["F2"] = 113] = "F2";
  Key10[Key10["F3"] = 114] = "F3";
  Key10[Key10["F4"] = 115] = "F4";
  Key10[Key10["F5"] = 116] = "F5";
  Key10[Key10["F6"] = 117] = "F6";
  Key10[Key10["F7"] = 118] = "F7";
  Key10[Key10["F8"] = 119] = "F8";
  Key10[Key10["F9"] = 120] = "F9";
  Key10[Key10["F10"] = 121] = "F10";
  Key10[Key10["F11"] = 122] = "F11";
  Key10[Key10["F12"] = 123] = "F12";
  Key10[Key10["NumLock"] = 144] = "NumLock";
  Key10[Key10["ScrollLock"] = 145] = "ScrollLock";
  Key10[Key10["Semicolon"] = 186] = "Semicolon";
  Key10[Key10["Equals"] = 187] = "Equals";
  Key10[Key10["Comma"] = 188] = "Comma";
  Key10[Key10["Dash"] = 189] = "Dash";
  Key10[Key10["Period"] = 190] = "Period";
  Key10[Key10["ForwardSlash"] = 191] = "ForwardSlash";
  Key10[Key10["GraveAccent"] = 192] = "GraveAccent";
  Key10[Key10["OpenBracket"] = 219] = "OpenBracket";
  Key10[Key10["BackSlash"] = 220] = "BackSlash";
  Key10[Key10["CloseBracket"] = 221] = "CloseBracket";
  Key10[Key10["SingleQuote"] = 222] = "SingleQuote";
  return Key10;
})(Key || {});
export {
  Key as K
};
