// @ownego/polaris-vue/dist/es/src/components/Layout/Layout.vue_vue_type_script_setup_true_lang.js
import { defineComponent, openBlock, createElementBlock, normalizeClass, unref, createBlock, withCtx, renderSlot } from "vue";
import { _ as _sfc_main$1 } from "./components/Section/Section.vue_vue_type_script_setup_true_lang.js";
const Layout = "Polaris-Layout";
const Section = "Polaris-Layout__Section";
const AnnotatedSection = "Polaris-Layout__AnnotatedSection";
const AnnotationWrapper = "Polaris-Layout__AnnotationWrapper";
const AnnotationContent = "Polaris-Layout__AnnotationContent";
const Annotation = "Polaris-Layout__Annotation";
const styles = {
  Layout,
  Section,
  "Section-fullWidth": "Polaris-Layout__Section--fullWidth",
  "Section-oneHalf": "Polaris-Layout__Section--oneHalf",
  "Section-oneThird": "Polaris-Layout__Section--oneThird",
  AnnotatedSection,
  AnnotationWrapper,
  AnnotationContent,
  Annotation
};
const _sfc_main = /* @__PURE__ */ defineComponent({
  __name: "Layout",
  props: {
    sectioned: { type: Boolean }
  },
  setup(__props) {
    return (_ctx, _cache) => {
      return openBlock(), createElementBlock("div", {
        class: normalizeClass(unref(styles).Layout)
      }, [
        _ctx.sectioned ? (openBlock(), createBlock(_sfc_main$1, { key: 0 }, {
          default: withCtx(() => [
            renderSlot(_ctx.$slots, "default")
          ]),
          _: 3
        })) : renderSlot(_ctx.$slots, "default", { key: 1 })
      ], 2);
    };
  }
});
export {
  _sfc_main as _,
  styles as s
};
