// @ownego/polaris-vue/dist/es/src/components/ColorPicker/components/AlphaPicker/AlphaPicker.vue_vue_type_script_setup_true_lang.js
import { defineComponent, ref, computed, onMounted, onBeforeUnmount, openBlock, createElementBlock, normalizeClass, unref, createElementVNode, normalizeStyle, createVNode } from "vue";
import { a as hsbToRgb } from "../../../../utilities/color-transformers.js";
import { _ as _sfc_main$1 } from "../Slidable/Slidable.vue_vue_type_script_setup_true_lang.js";
import { c as calculateDraggerY, a as alphaForDraggerY } from "../../../../../polaris/polaris-react/src/components/ColorPicker/components/AlphaPicker/utilities.js";
import { s as styles } from "../../ColorPicker.vue_vue_type_script_setup_true_lang.js";
const _sfc_main = /* @__PURE__ */ defineComponent({
  __name: "AlphaPicker",
  props: {
    color: {},
    alpha: {}
  },
  emits: ["change"],
  setup(__props, { emit: __emit }) {
    const props = __props;
    const emits = __emit;
    const sliderHeight = ref(0);
    const draggerHeight = ref(0);
    const alphaPickerRef = ref(null);
    const observer = ref(null);
    const draggerY = computed(() => {
      return calculateDraggerY(props.alpha, sliderHeight.value, draggerHeight.value);
    });
    const background = computed(() => {
      return alphaGradientForColor(props.color);
    });
    onMounted(() => {
      if (!alphaPickerRef.value) {
        return;
      }
      observer.value = new ResizeObserver(() => {
        setSliderHeight();
      });
      observer.value.observe(alphaPickerRef.value);
      setSliderHeight();
    });
    onBeforeUnmount(() => {
      var _a;
      (_a = observer.value) == null ? void 0 : _a.disconnect();
    });
    const setSliderHeight = () => {
      if (alphaPickerRef.value) {
        sliderHeight.value = alphaPickerRef.value.clientHeight;
      }
    };
    const setDraggerHeight = (height) => {
      draggerHeight.value = height;
    };
    const handleChange = ({ y }) => {
      const alpha = alphaForDraggerY(y, sliderHeight.value);
      emits("change", alpha);
    };
    function alphaGradientForColor(color) {
      const { red, green, blue } = hsbToRgb(color);
      const rgb = `${red}, ${green}, ${blue}`;
      return `linear-gradient(to top, rgba(${rgb}, 0) 18px, rgba(${rgb}, 1) calc(100% - 18px))`;
    }
    return (_ctx, _cache) => {
      return openBlock(), createElementBlock("div", {
        class: normalizeClass(unref(styles).AlphaPicker),
        ref_key: "alphaPickerRef",
        ref: alphaPickerRef
      }, [
        createElementVNode("div", {
          class: normalizeClass(unref(styles).ColorLayer),
          style: normalizeStyle({ background: background.value })
        }, null, 6),
        createVNode(unref(_sfc_main$1), {
          draggerY: draggerY.value,
          draggerX: 0,
          onChange: handleChange,
          onDraggerHeight: setDraggerHeight
        }, null, 8, ["draggerY"])
      ], 2);
    };
  }
});
export {
  _sfc_main as _
};
