// @ownego/polaris-vue/dist/es/src/components/Page/components/Header/Header.vue_vue_type_script_setup_true_lang.js
import { defineComponent, useSlots, computed, provide, openBlock, createBlock, unref, withCtx, createElementBlock, createVNode, createTextVNode, toDisplayString, createCommentVNode, createElementVNode, normalizeClass, normalizeProps, guardReactiveProps, renderSlot, mergeProps, Fragment } from "vue";
import { u as useI18n } from "../../../../use/useI18n.js";
import { u as useHasSlot } from "../../../../use/useHasSlot.js";
import { c as classNames } from "../../../../utilities/css.js";
import { u as useMediaQueryContext } from "../../../../use/useMediaQuery.js";
import { _ as _sfc_main$1 } from "../../../Box/Box.vue_vue_type_script_setup_true_lang.js";
import { _ as _sfc_main$3 } from "../../../Breadcrumbs/Breadcrumbs.vue_vue_type_script_setup_true_lang.js";
import { _ as _sfc_main$8 } from "../../../InlineStack/InlineStack.vue_vue_type_script_setup_true_lang.js";
import { _ as _sfc_main$6 } from "./children/PrimaryAction.vue_vue_type_script_setup_true_lang.js";
import { _ as _sfc_main$5 } from "./children/ActionMenu.vue_vue_type_script_setup_true_lang.js";
import { _ as _sfc_main$7 } from "./children/Pagination.vue_vue_type_script_setup_true_lang.js";
import { _ as _sfc_main$4 } from "./children/PageTitle.vue_vue_type_script_setup_true_lang.js";
import { _ as _sfc_main$2 } from "../../../Text/Text.vue_vue_type_script_setup_true_lang.js";
const TitleWrapper = "Polaris-Page-Header__TitleWrapper";
const TitleWrapperExpand = "Polaris-Page-Header__TitleWrapperExpand";
const BreadcrumbWrapper = "Polaris-Page-Header__BreadcrumbWrapper";
const PaginationWrapper = "Polaris-Page-Header__PaginationWrapper";
const PrimaryActionWrapper = "Polaris-Page-Header__PrimaryActionWrapper";
const Row = "Polaris-Page-Header__Row";
const mobileView = "Polaris-Page-Header--mobileView";
const RightAlign = "Polaris-Page-Header__RightAlign";
const noBreadcrumbs = "Polaris-Page-Header--noBreadcrumbs";
const AdditionalMetaData = "Polaris-Page-Header__AdditionalMetaData";
const Actions = "Polaris-Page-Header__Actions";
const longTitle = "Polaris-Page-Header--longTitle";
const mediumTitle = "Polaris-Page-Header--mediumTitle";
const isSingleRow = "Polaris-Page-Header--isSingleRow";
const styles = {
  TitleWrapper,
  TitleWrapperExpand,
  BreadcrumbWrapper,
  PaginationWrapper,
  PrimaryActionWrapper,
  Row,
  mobileView,
  RightAlign,
  noBreadcrumbs,
  AdditionalMetaData,
  Actions,
  longTitle,
  mediumTitle,
  isSingleRow
};
const _hoisted_1 = {
  key: 0,
  role: "status"
};
const SHORT_TITLE = 20;
const REALLY_SHORT_TITLE = 8;
const LONG_TITLE = 34;
const _sfc_main = /* @__PURE__ */ defineComponent({
  __name: "Header",
  props: {
    titleHidden: { type: Boolean },
    pageReadyAccessibilityLabel: {},
    filterActions: { type: Boolean },
    primaryAction: {},
    pagination: {},
    backAction: {},
    secondaryActions: {},
    actionGroups: {},
    additionalMetadata: {},
    title: {},
    subtitle: {},
    compactTitle: { type: Boolean },
    hasSubtitleMaxWidth: { type: Boolean }
  },
  emits: ["action-rollup"],
  setup(__props, { emit: __emit }) {
    const props = __props;
    const i18n = useI18n();
    const { isNavigationCollapsed } = useMediaQueryContext();
    const slots = useSlots();
    const { hasSlot } = useHasSlot();
    const isSingleRow2 = computed(
      () => {
        var _a, _b;
        return (!props.primaryAction || hasSlot(slots.primaryAction)) && !props.pagination && (!((_a = props.secondaryActions) == null ? void 0 : _a.length) || hasSlot(slots.secondaryActions)) && !((_b = props.actionGroups) == null ? void 0 : _b.length);
      }
    );
    const hasActionGroupsOrSecondaryActions = computed(
      () => props.actionGroups && props.actionGroups.length > 0 || props.secondaryActions && props.secondaryActions.length > 0 || hasSlot(slots.secondaryActions)
    );
    const headerClassNames = computed(
      () => classNames(
        isSingleRow2 && styles.isSingleRow,
        (props.backAction || props.pagination && !isNavigationCollapsed) && styles.hasNavigation,
        (props.secondaryActions && props.secondaryActions.length > 0 || hasGroupsWithActions(props.actionGroups) || hasSlot(slots.secondaryActions)) && styles.hasActionMenu,
        isNavigationCollapsed && styles.mobileView,
        !props.backAction && styles.noBreadcrumbs,
        props.title && props.title.length < LONG_TITLE && styles.mediumTitle,
        props.title && props.title.length > LONG_TITLE && styles.longTitle
      )
    );
    const labelForPageReadyAccessibilityLabel = computed(
      () => props.pageReadyAccessibilityLabel || props.title
    );
    const conditionMobileCompact = computed(
      () => isNavigationCollapsed && !props.backAction && props.title != null && props.title.length <= REALLY_SHORT_TITLE
    );
    const conditionDesktopCompact = computed(() => {
      return !isNavigationCollapsed && !props.pagination && !props.secondaryActions && !slots.secondaryActions && props.title != null && props.title.length <= SHORT_TITLE;
    });
    const primaryActionProps = computed(() => {
      return { ...props.primaryAction };
    });
    const pageTitleProps = computed(() => {
      return {
        title: props.title,
        subtitle: props.subtitle,
        compactTitle: props.compactTitle,
        hasSubtitleMaxWidth: hasActionGroupsOrSecondaryActions.value
      };
    });
    const actionMenuProps = computed(() => {
      return {
        title: props.title,
        secondaryActions: props.secondaryActions,
        actionGroups: props.actionGroups
      };
    });
    const conditionSlot4 = computed(() => {
      return (isNavigationCollapsed || conditionDesktopCompact.value) && primaryActionProps.value || !isNavigationCollapsed && !conditionDesktopCompact.value && props.pagination;
    });
    const conditionHasRightAlign = computed(() => {
      return props.primaryAction || slots.primaryAction || (props.secondaryActions || slots.secondaryActions) && (props.secondaryActions && !slots.secondaryActions && props.secondaryActions.length > 0 || slots.secondaryActions && !props.secondaryActions && !slots.secondaryActions.length) || (props.actionGroups && props.actionGroups.length > 0 || slots.actionGroups);
    });
    const hasGroupsWithActions = (groups = []) => {
      return groups.length === 0 ? false : groups.some((group) => group.actions.length > 0);
    };
    provide("filterActions", Boolean(props.filterActions));
    return (_ctx, _cache) => {
      return openBlock(), createBlock(unref(_sfc_main$1), {
        position: "relative",
        visuallyHidden: _ctx.titleHidden,
        paddingBlockStart: { xs: "400", md: "600" },
        paddingBlockEnd: { xs: "400", md: "600" },
        paddingInlineStart: { xs: "400", sm: "0" },
        paddingInlineEnd: { xs: "400", sm: "0" }
      }, {
        default: withCtx(() => [
          labelForPageReadyAccessibilityLabel.value ? (openBlock(), createElementBlock("div", _hoisted_1, [
            createVNode(unref(_sfc_main$2), {
              visuallyHidden: "",
              as: "p"
            }, {
              default: withCtx(() => [
                createTextVNode(toDisplayString(unref(i18n).translate(
                  "Polaris.Page.Header.pageReadyAccessibilityLabel",
                  { title: labelForPageReadyAccessibilityLabel.value }
                )), 1)
              ]),
              _: 1
            })
          ])) : createCommentVNode("", true),
          createElementVNode("div", {
            class: normalizeClass(headerClassNames.value)
          }, [
            createElementVNode("div", {
              class: normalizeClass(unref(styles).Row)
            }, [
              !conditionMobileCompact.value ? (openBlock(), createElementBlock("div", {
                key: 0,
                class: normalizeClass(unref(styles).BreadcrumbWrapper)
              }, [
                createVNode(unref(_sfc_main$1), {
                  maxWidth: "100%",
                  printHidden: "",
                  paddingInlineEnd: "100"
                }, {
                  default: withCtx(() => [
                    _ctx.backAction ? (openBlock(), createBlock(unref(_sfc_main$3), {
                      key: 0,
                      backAction: _ctx.backAction
                    }, null, 8, ["backAction"])) : createCommentVNode("", true)
                  ]),
                  _: 1
                })
              ], 2)) : createCommentVNode("", true),
              createVNode(unref(_sfc_main$4), normalizeProps(guardReactiveProps(pageTitleProps.value)), {
                default: withCtx(() => [
                  renderSlot(_ctx.$slots, "pageTitle")
                ]),
                _: 3
              }, 16),
              conditionHasRightAlign.value ? (openBlock(), createElementBlock("div", {
                key: 1,
                class: normalizeClass(unref(styles).RightAlign)
              }, [
                createElementVNode("div", {
                  class: normalizeClass(unref(styles).Actions)
                }, [
                  createVNode(unref(_sfc_main$5), normalizeProps(guardReactiveProps(actionMenuProps.value)), {
                    default: withCtx(() => [
                      renderSlot(_ctx.$slots, "secondaryActions")
                    ]),
                    _: 3
                  }, 16),
                  (unref(slots).primaryAction || _ctx.primaryAction) && !unref(isNavigationCollapsed) && !conditionDesktopCompact.value ? (openBlock(), createBlock(unref(_sfc_main$6), normalizeProps(mergeProps({ key: 0 }, primaryActionProps.value)), {
                    default: withCtx(() => [
                      renderSlot(_ctx.$slots, "primaryAction")
                    ]),
                    _: 3
                  }, 16)) : createCommentVNode("", true),
                  conditionSlot4.value ? (openBlock(), createElementBlock(Fragment, { key: 1 }, [
                    (unref(slots).primaryAction || _ctx.primaryAction) && (unref(isNavigationCollapsed) || conditionDesktopCompact.value) ? (openBlock(), createBlock(unref(_sfc_main$6), normalizeProps(mergeProps({ key: 0 }, primaryActionProps.value)), {
                      default: withCtx(() => [
                        renderSlot(_ctx.$slots, "primaryAction")
                      ]),
                      _: 3
                    }, 16)) : createCommentVNode("", true),
                    !unref(isNavigationCollapsed) && !conditionDesktopCompact.value ? (openBlock(), createBlock(unref(_sfc_main$7), {
                      key: 1,
                      pagination: _ctx.pagination
                    }, {
                      default: withCtx(() => [
                        unref(hasSlot)(unref(slots).pagination) ? renderSlot(_ctx.$slots, "pagination", { key: 0 }) : createCommentVNode("", true)
                      ]),
                      _: 3
                    }, 8, ["pagination"])) : createCommentVNode("", true)
                  ], 64)) : createCommentVNode("", true)
                ], 2)
              ], 2)) : createCommentVNode("", true)
            ], 2),
            _ctx.additionalMetadata ? (openBlock(), createElementBlock("div", {
              key: 0,
              class: normalizeClass(unref(styles).Row)
            }, [
              createVNode(unref(_sfc_main$8), { gap: "400" }, {
                default: withCtx(() => [
                  createElementVNode("div", {
                    class: normalizeClass(unref(styles).AdditionalMetaData)
                  }, [
                    createVNode(unref(_sfc_main$2), {
                      tone: "subdued",
                      as: "span",
                      variant: "bodySm"
                    }, {
                      default: withCtx(() => [
                        renderSlot(_ctx.$slots, "additionalMetadata")
                      ]),
                      _: 3
                    })
                  ], 2)
                ]),
                _: 3
              })
            ], 2)) : createCommentVNode("", true)
          ], 2)
        ]),
        _: 3
      }, 8, ["visuallyHidden"]);
    };
  }
});
export {
  _sfc_main as _,
  styles as s
};
