// @ownego/polaris-vue/dist/es/src/components/Tabs/components/CreateViewModal/CreateViewModal.vue_vue_type_script_setup_true_lang.js
import { defineComponent, useSlots, ref, computed, watch, openBlock, createBlock, unref, withCtx, createTextVNode, toDisplayString, renderSlot, createVNode, createElementVNode } from "vue";
import { u as useI18n } from "../../../../use/useI18n.js";
import { u as useIsTouchDevice } from "../../../../utilities/use-is-touch-device.js";
import { f as focusFirstFocusableNode } from "../../../../utilities/focus.js";
import { _ as _sfc_main$3 } from "../../../Form/Form.vue_vue_type_script_setup_true_lang.js";
import { _ as _sfc_main$4 } from "../../../FormLayout/FormLayout.vue_vue_type_script_setup_true_lang.js";
import { _ as _sfc_main$1 } from "../../../Modal/Modal.vue_vue_type_script_setup_true_lang.js";
import { _ as _sfc_main$2 } from "../../../Modal/components/Section/Section.vue_vue_type_script_setup_true_lang.js";
import { _ as _sfc_main$5 } from "../../../TextField/TextField.vue_vue_type_script_setup_true_lang.js";
const MAX_VIEW_NAME_LENGTH = 40;
const _sfc_main = /* @__PURE__ */ defineComponent({
  __name: "CreateViewModal",
  props: {
    open: { type: Boolean },
    viewNames: {}
  },
  emits: ["close", "click-primary-action", "click-secondary-action", "update:modelValue"],
  setup(__props, { emit: __emit }) {
    const props = __props;
    useSlots();
    const emits = __emit;
    const i18n = useI18n();
    const isTouchDevice = useIsTouchDevice();
    const modalValue = ref("");
    const model = computed({
      get() {
        return "";
      },
      set(value) {
        modalValue.value = value;
        emits("update:modelValue", value);
      }
    });
    const loading = ref(false);
    const container = ref(null);
    const hasSameNameError = computed(() => props.viewNames.some(
      (viewName) => viewName.trim().toLowerCase() === modalValue.value.trim().toLowerCase()
    ));
    const isPrimaryActionDisabled = computed(
      () => !modalValue.value || hasSameNameError.value || loading.value || modalValue.value.length > MAX_VIEW_NAME_LENGTH
    );
    const primaryAction = computed(() => {
      return {
        content: i18n.translate("Polaris.Tabs.CreateViewModal.create"),
        disabled: isPrimaryActionDisabled.value,
        onAction: handlePrimaryAction
      };
    });
    const secondaryActions = computed(() => [
      {
        content: i18n.translate("Polaris.Tabs.CreateViewModal.cancel"),
        onAction: handleSecondaryAction
      }
    ]);
    const errorMessage = computed(
      () => hasSameNameError.value && i18n.translate("Polaris.Tabs.CreateViewModal.errors.sameName", { name: modalValue.value })
    );
    const handlePrimaryAction = async () => {
      if (hasSameNameError.value || isPrimaryActionDisabled.value) {
        return;
      }
      loading.value = true;
      await emits("click-primary-action", modalValue.value);
      loading.value = false;
      model.value = "";
      emits("close");
    };
    const handleSecondaryAction = () => {
      emits("click-secondary-action");
      model.value = "";
      emits("close");
    };
    watch(
      () => [props.open, isTouchDevice.value],
      () => {
        if (!container.value || isTouchDevice.value) return;
        if (props.open) {
          focusFirstFocusableNode(container.value);
          const timeout = setTimeout(() => {
            if (!container.value) return;
            focusFirstFocusableNode(container.value);
          }, 50);
          return () => clearTimeout(timeout);
        }
      },
      {
        flush: "post"
      }
    );
    return (_ctx, _cache) => {
      return openBlock(), createBlock(unref(_sfc_main$1), {
        open: _ctx.open,
        primaryAction: primaryAction.value,
        secondaryActions: secondaryActions.value,
        onClose: _cache[1] || (_cache[1] = ($event) => emits("close"))
      }, {
        title: withCtx(() => [
          createTextVNode(toDisplayString(unref(i18n).translate("Polaris.Tabs.CreateViewModal.title")), 1)
        ]),
        activator: withCtx(() => [
          renderSlot(_ctx.$slots, "default")
        ]),
        default: withCtx(() => [
          createVNode(unref(_sfc_main$2), null, {
            default: withCtx(() => [
              createVNode(unref(_sfc_main$3), { onSubmit: handlePrimaryAction }, {
                default: withCtx(() => [
                  createVNode(unref(_sfc_main$4), null, {
                    default: withCtx(() => [
                      createElementVNode("div", {
                        ref_key: "container",
                        ref: container
                      }, [
                        createVNode(unref(_sfc_main$5), {
                          autoComplete: "off",
                          modelValue: model.value,
                          "onUpdate:modelValue": _cache[0] || (_cache[0] = ($event) => model.value = $event),
                          label: unref(i18n).translate("Polaris.Tabs.CreateViewModal.label"),
                          maxLength: MAX_VIEW_NAME_LENGTH,
                          showCharacterCount: true,
                          error: errorMessage.value
                        }, null, 8, ["modelValue", "label", "error"])
                      ], 512)
                    ]),
                    _: 1
                  })
                ]),
                _: 1
              })
            ]),
            _: 1
          })
        ]),
        _: 3
      }, 8, ["open", "primaryAction", "secondaryActions"]);
    };
  }
});
export {
  _sfc_main as _
};
