// @ownego/polaris-vue/dist/es/src/components/DatePicker/components/Weekday/Weekday.vue_vue_type_script_setup_true_lang.js
import { defineComponent, openBlock, createElementBlock, normalizeClass, unref, createVNode, withCtx, createTextVNode, toDisplayString } from "vue";
import { s as styles } from "../../DatePicker.vue_vue_type_script_setup_true_lang.js";
import { _ as _sfc_main$1 } from "../../../Text/Text.vue_vue_type_script_setup_true_lang.js";
const _hoisted_1 = ["aria-label"];
const _sfc_main = /* @__PURE__ */ defineComponent({
  __name: "Weekday",
  props: {
    label: {},
    title: {},
    current: { type: Boolean }
  },
  setup(__props) {
    return (_ctx, _cache) => {
      return openBlock(), createElementBlock("th", {
        "aria-label": _ctx.label,
        scope: "col",
        class: normalizeClass(unref(styles).Weekday)
      }, [
        createVNode(unref(_sfc_main$1), {
          as: "span",
          variant: "bodySm",
          alignment: "center",
          "font-weight": _ctx.current ? "bold" : "regular",
          tone: !_ctx.current ? "subdued" : void 0
        }, {
          default: withCtx(() => [
            createTextVNode(toDisplayString(_ctx.title), 1)
          ]),
          _: 1
        }, 8, ["font-weight", "tone"])
      ], 10, _hoisted_1);
    };
  }
});
export {
  _sfc_main as _
};
